hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/JpaMetamodelImpl.java

```diff
@@ -272,37 +289,32 @@ public void visitEmbeddables(Consumer<EmbeddableDomainType<?>> action) {
 
 	@Override
 	public String qualifyImportableName(String queryName) {
-		return resolveImportedName( queryName );
+		final ImportInfo<?> importInfo = resolveImport( queryName );
+		return importInfo == null ? null : importInfo.importedName;
 	}
 
-	private String resolveImportedName(String name) {
-		String result = nameToImportNameMap.get( name );
-		if ( result == null ) {
+	private <T> ImportInfo<T> resolveImport(String name) {
+		//noinspection unchecked
+		final ImportInfo<T> result = (ImportInfo<T>) nameToImportMap.computeIfAbsent( name, unknownName -> {
```

Comment:
The benefit of using `ConcurrentHashMap#computeIfAbsent()` is thread safety and atomicity.